SharpPcap/LibPcap/LibPcapLiveDevice.cs

```diff
@@ -156,67 +180,89 @@ public override void Open(DeviceConfiguration configuration)
-                            nameof(configuration.Immediate), 
-                            (int)PcapError.PlatformNotSupported
-                        );
+                        // We got authentication, so this is an rpcap device
+                        var auth = RemoteAuthentication.CreateAuth(credentials);
+                        Handle = LibPcapSafeNativeMethods.pcap_open(
+                            Name,                               // name of the device
+                            configuration.Snaplen,              // portion of the packet to capture.
+                            (short)mode,                        // flags
+                            (short)configuration.ReadTimeout,   // read timeout
+                            ref auth,                           // authentication
+                            errbuf);                            // error buffer
                     }
-                    else if (immediate_supported)
+                    else
                     {
-                        var immediate = configuration.Immediate.Value ? 1 : 0;
-                        Configure(
-                            configuration, nameof(configuration.Immediate),
-                            LibPcapSafeNativeMethods.pcap_set_immediate_mode, immediate
-                        );
+                        //We got no authentication, but modes other than 
+                        // DeviceModes.Promiscuous and DeviceModes.MaxResponsiveness require pcap_open()
+                        Handle = LibPcapSafeNativeMethods.pcap_open(
+                            Name,                               // name of the device
+                            configuration.Snaplen,              // portion of the packet to capture.
+                            (short)mode,                        // flags
+                            (short)configuration.ReadTimeout,   // read timeout
+                            IntPtr.Zero,                        // authentication
+                            errbuf);                            // error buffer
                     }
                 }
 
-                var activationResult = LibPcapSafeNativeMethods.pcap_activate(Handle);
-                if (activationResult < 0)
+                if (Handle.IsInvalid)
                 {
-                    string err = "Unable to activate the adapter (" + Name + "). Return code: " + activationResult.ToString();
+                    string err = "Unable to open the adapter (" + Name + "). " + errbuf.ToString();
                     throw new PcapException(err);
                 }
+
+                // Is there a better check for an activated handle?
+                // This check breaks some unit test using pcap_geterror().
+                //var actStatus = LibPcapSafeNativeMethods.pcap_activate(Handle);
+                //Debug.Assert(LibPcapSafeNativeMethods.PCAP_ERROR_ACTIVATED == actStatus || actStatus >= 0);
```

Comment:
A device created using `pcap_open` is already activated, you can not perform pre activation configuration on it.